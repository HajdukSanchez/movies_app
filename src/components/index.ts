export { MovieList } from './movieList/MovieList';
export { MoviePoster } from './moviePoster/MoviePoster';
export { Loading } from './loading/Loading';
export { MovieDetails } from './movieDetails/MovieDetails';
export { ActorCard } from './actorCard/ActorCard';
