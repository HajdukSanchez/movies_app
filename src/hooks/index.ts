export { useMovies } from './useMovies/useMovies';
export { useMovieDetails } from './useMovieDetails/useMovieDetails';
