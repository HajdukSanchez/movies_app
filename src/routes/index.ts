/**
 * Params for Sack navigation.
 */
export type RootStackParamList = {
  Home: undefined;
  Detail: undefined;
};
