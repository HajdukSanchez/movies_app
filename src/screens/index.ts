export { DetailScreen } from './detailScreen/DetailScreen';
export { HomeScreen } from './homeScreen/HomeScreen';
