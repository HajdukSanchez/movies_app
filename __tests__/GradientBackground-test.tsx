import React from 'react';
import { Text } from 'react-native';
import { render, screen } from '@testing-library/react-native';

import { GradientBackground } from '../src/components';

describe('<GradientBackground/>', () => {
  let child: JSX.Element;
  let children: JSX.Element[];

  beforeAll(() => {
    child = <Text>Hello</Text>;
    children = [child, child, child];
  });

  test('should render correctly with one child', () => {
    const bg = render(<GradientBackground children={child} />).toJSON();
    expect(bg).toMatchSnapshot();
    expect(screen.getByText('Hello')).toBeTruthy();
  });

  test('should render correctly with children', () => {
    const bg = render(<GradientBackground children={children} />).toJSON();
    expect(bg).toMatchSnapshot();
    const items = screen.getAllByText('Hello');
    expect(items).toBeTruthy();
    expect(items).toHaveLength(3);
  });
});
