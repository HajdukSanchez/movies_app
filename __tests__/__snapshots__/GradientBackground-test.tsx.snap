// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GradientBackground/> should render correctly with children 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <BVLinearGradient
    colors={
      Array [
        4284161823,
        4283968813,
        4290877849,
      ]
    }
    endPoint={
      Object {
        "x": 0.8,
        "y": 0.8,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <Text>
      Hello
    </Text>
    <Text>
      Hello
    </Text>
    <Text>
      Hello
    </Text>
  </BVLinearGradient>
</View>
`;

exports[`<GradientBackground/> should render correctly with one child 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <BVLinearGradient
    colors={
      Array [
        4284161823,
        4283968813,
        4290877849,
      ]
    }
    endPoint={
      Object {
        "x": 0.8,
        "y": 0.8,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <Text>
      Hello
    </Text>
  </BVLinearGradient>
</View>
`;
